### ISSUE TITLE
Build: Should we use `stable` or changing versions in our build tools?

### ISSUE THREAD

**Issue by slim-bean** [grafana/loki#4708]
Over the weekend a new version of `shellcheck` was released and our build step uses `stable` as the image tag, this meant that suddenly and unexpectedly all current PR builds started to fail as the new version of shellcheck found errors in existing scripts.

I'm of two minds about this, I think we should pin the versions of tooling in our build to stabilize builds, however I realize then that we would likely infrequently (never?) update tools like this...

Creating this issue to start a discussion.

**Comment by dannykopping** [comment]
Seems this is limited to `shellcheck` only, right? Maybe we should make the title specific to this?

```bash
$ grep image: .drone/drone.yml | sort | uniq 
  image: alpine
  image: cstyan/ecr
  image: golang:windowsservercore-1809
  image: grafana/loki-build-image:0.18.0
  image: koalaman/shellcheck-alpine:stable
  image: plugins/docker
  image: plugins/manifest
  image: us.gcr.io/kubernetes-dev/drone/plugins/deploy-image
```

I'd support setting a fixed version of `shellcheck`; I don't think that shell recommendations change too often. AFAIK it doesn't do any security scanning either.

**Comment by DylanGuedes** [comment]
What is the severity of the reported errors? If they are relevant I think keeping `stable` has more upsides than downsides.

**Comment by cyriltovena** [comment]
I'm on the side of pining dependencies to avoid unexpected breaking change.

**Comment by james-callahan** [comment]
> I don't think that shell recommendations change too often. AFAIK it doesn't do any security scanning either.

They do though. that's the reason this issue is open.
I recommend leaving it un-pinned, otherwise you'll never upgrade.

However I might suggest making the CI step optional so that you can merge despite it failing.

**Comment by jeschkies** [comment]
> I think we should pin the versions of tooling in our build to stabilize builds

Yes, we should always pin our build tool version. Upgrades are simply enforce when someone needs a newer version. I wouldn't worry about it too much.

**Comment by JStickler** [comment]
Closing this issue as fixed due to the migration from Drone CI to GitHub Actions. Shellcheck is now installed via a pinned version in the GitHub Actions workflow, and the .drone/ CI configuration no longer exists. Please reopen if there are other build tools with unpinned versions that concern you.
